app/api/convert/route.ts

```diff
@@ -0,0 +1,63 @@
+import fs from "fs";
+import { exec } from "child_process";
+import path from "path";
+
+// Directory to store temporary files
+const tempDir = path.join(__dirname, "temp");
+if (!fs.existsSync(tempDir)) {
+  fs.mkdirSync(tempDir, { recursive: true });
+}
+
+/**
+ * Function to convert a base64 JP2 file to PNG and return the PNG as base64.
+ * @param {string} base64JP2 - JP2 image in base64.
+ * @returns {Promise<string>} - PNG image in base64.
+ */
+const convertBase64JP2ToPNGBase64 = (base64JP2: string) => {
+  return new Promise((resolve, reject) => {
+    try {
+      const jp2Filename = `input_${Date.now()}.jp2`;
+      const jp2Path = path.join(tempDir, jp2Filename);
+      const pngFilename = `output_${Date.now()}.png`;
+      const pngPath = path.join(tempDir, pngFilename);
+      const jp2Buffer = Buffer.from(base64JP2, "base64");
+      // Save the JP2 file
+      fs.writeFileSync(jp2Path, jp2Buffer);
+      // Execute the conversion using ImageMagick
+      const command = `convert "${jp2Path}" "${pngPath}"`;
+      exec(command, (err, stdout, stderr) => {
+        if (err) {
+          // Delete temporary files
+          fs.unlinkSync(jp2Path);
```

Comment:
JP2 file will be always present and we want to delete it as soon as possible, so maybe you can put `          fs.unlinkSync(jp2Path);` in the first line of the exec callback.